### PR TITLE
OSDP - Don't use TEST_RANDOM_GENERATOR

### PR DESCRIPTION
**Pull request by ceolin**
- Do not use TEST_RANDOM_GENERATOR
- Do not select unecessarily ENTROPY_GENERATOR
- Do not select wrong SERIAL options

### CODE REVIEW

File: samples/subsys/mgmt/osdp/sample.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+tests:
+  sample.subsys.mgmt.osdp:
+    tags: osdp
+    filter: dt_compat_enabled_with_alias("gpio-leds", "led0")
+    harness: osdp
+    integration_platforms:
+      - frdm_k64f
+
+    # FIXME: exclude due to build error
+    platform_exclude: qemu_xtensa
```
Comment (nashif) [inline samples/subsys/mgmt/osdp/sample.yaml:12]: look at this

https://github.com/zephyrproject-rtos/zephyr/pull/27913#issuecomment-685153646

Comment (ceolin) [reply samples/subsys/mgmt/osdp/sample.yaml]: cool, just remove that commit.

Comment (jenmwms) [reply samples/subsys/mgmt/osdp/sample.yaml]: Thanks! I'll focus on osdp/sample.yaml there (#27913). For Kconfig, the comment^^ mentioned to also remove `SERIAL_HAS_DRIVER` and `UART_INTERRUPT_DRIVEN`. Do we need that too?

Comment (ceolin) [reply samples/subsys/mgmt/osdp/sample.yaml]: I haven't checked that to be honest. Will look

**Review by jenmwms — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**